torchtitan/components/checkpoint.py

```diff
@@ -430,10 +521,17 @@ def load(self, step: int = -1) -> bool:
                     f"--checkpoint.load_step={step} but checkpoint {checkpoint_id} is not found."
                 )
 
+        checkpoint_type = self._find_checkpoint_type(checkpoint_id)
+        if checkpoint_type == CheckpointType.SAFETENSORS:
+            model_only = True
```

Comment:
We should assert if `model_only` is not True, rather than silently change `model_only` value.